env0/resource_agent_project_assignment.go

```diff
@@ -0,0 +1,196 @@
+	}
+
+	apaLock.Lock()
+	defer apaLock.Unlock()
+
+	apiClient := meta.(client.ApiClientInterface)
+	assignments, err := apiClient.ProjectsAgentsAssignments()
+	if err != nil {
+		return diag.Errorf("could not get project-agent assignments: %v", err)
+	}
+
+	for projectId, agentId := range assignments.ProjectsAgents {
+		if projectId == newAssignment.ProjectId && agentId == newAssignment.AgentId {
+			return diag.Errorf("assignment for project id %v and agent id %v already exist", projectId, agentId)
+		}
+	}
+
+	assignments.ProjectsAgents[newAssignment.ProjectId] = newAssignment.AgentId
+
+	if _, err := apiClient.AssignAgentsToProjects(assignments.ProjectsAgents); err != nil {
+		return diag.Errorf("could not update project-agent assignments: %v", err)
+	}
+
+	d.SetId(newAssignment.GetId())
+
+	return nil
+}
+
+func resourceAgentProjectAssignmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	apaLock.Lock()
+	defer apaLock.Unlock()
+
+	assignments, err := apiClient.ProjectsAgentsAssignments()
+	if err != nil {
+		return diag.Errorf("could not get project-agent assignments: %v", err)
+	}
+
+	var assignment *AgentProjectAssignment
+	for projectId, agentId := range assignments.ProjectsAgents {
+		if d.Id() == GetAgentProjectAssignmentId(agentId.(string), projectId) {
+			assignment = &AgentProjectAssignment{
+				AgentId:   agentId.(string),
+				ProjectId: projectId,
+			}
+			break
+		}
+	}
+
+	if assignment == nil {
+		log.Printf("[WARN] Drift Detected: Terraform will remove %s from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err := writeResourceData(assignment, d); err != nil {
+		diag.Errorf("schema resource data serialization failed: %v", err)
+	}
+
+	return nil
+}
+
+func resourceAgentProjectAssignmentDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	projectId := d.Get("project_id").(string)
+	agentId := d.Get("agent_id").(string)
+
+	apaLock.Lock()
+	defer apaLock.Unlock()
+
+	apiClient := meta.(client.ApiClientInterface)
+
+	assignments, err := apiClient.ProjectsAgentsAssignments()
+	if err != nil {
+		return diag.Errorf("could not get project-agent assignments: %v", err)
+	}
+
+	newAssignments := make(map[string]interface{})
+
+	// Remove from the assignments the deleted assignment.
+	for projectIdOther, agentIdOther := range assignments.ProjectsAgents {
+		if projectId != projectIdOther || agentId != agentIdOther {
+			newAssignments[projectIdOther] = agentIdOther
+		}
+	}
+
+	if _, err := apiClient.AssignAgentsToProjects(newAssignments); err != nil {
+		return diag.Errorf("could not update project-agent assignments: %v", err)
+	}
+
+	return nil
+}
+
+func resourceAgentProjectAssignmentImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	ss := strings.Split(d.Id(), "_")
```

Comment:
rename please, agent_project is more readable

Reply:
@sagilaufer1992 , sorry: rename what?

Reply:
`ss` is not a good name IMO... the other abbreviations in the function signature seem more like a convention that repeat in our code already

Reply:
@sagilaufer1992 
https://github.com/env0/terraform-provider-env0/pull/325/commits/80a25e23a96c0a92a23c74a8cda3c6ab7d3d30e1